projects/packages/jetpack-mu-wpcom/src/features/wpcom-sidebar-notice/wpcom-sidebar-notice.php

```diff
@@ -75,14 +82,17 @@ function wpcom_get_sidebar_notice() {
 		$message = $jitm->get_messages( $message_path, wp_json_encode( array( 'message_path' => $message_path ) ), false );
 	}
 
+	$cache_loaded = true;
+
 	if ( ! isset( $message[0] ) ) {
+		$cached_notice = null;
 		return null;
 	}
 
 	// Serialize message as object (on Simple sites we have an array, on Atomic sites we have an object).
 	$message = json_decode( wp_json_encode( $message[0] ) );
```

Comment:
`json_decode` can throw, which means its technically possible for `$cache_loaded` to get set to `true` when `$cached_notice` hasn't been initialised.

It's duplication, but I think its more technically correct to set `$cache_loaded = true` in both places `$cache_notice` gets set.

Either that or use a sentinel value like:

```php
static $cached_notice = 'UNSET';

if ( $cached_notice !== 'UNSET' ) {
	return $cached_notice;
}
```

Reply:
I also like the idea of aligning `$cache_loaded` with `$cache_notice` (or using the sentinel value approach) to make things a bit more robust, but I don't think it matters in practice. 

If `json_decode` throws an error, we'd be returning `$cached_notice = null` which is maybe better than trying again and likely throwing another error?

Anyway, these errors are something that we don't seem to currently handle, so probably not something to worry about in the scope of this PR.